### PR TITLE
handle smaller sync responses introduced in synapse 1.38.0

### PR DESCRIPTION
**Pull request by surakin**
Synapse 1.38.0 introduced smaller sync responses (https://github.com/matrix-org/synapse/pull/10214) by removing empty fields, but this library did not handle it correctly.
This broke the Home Assistant matrix integration, for example.

### CODE REVIEW

File: matrix_client/client.py [hunk 1]
```diff
@@ -583,60 +583,68 @@ def _sync(self, timeout_ms=30000):
...
-                event['room_id'] = room_id
-                room._put_ephemeral_event(event)
-
-                for listener in self.ephemeral_listeners:
-                    if (
-                        listener['event_type'] is None or
-                        listener['event_type'] == event['type']
-                    ):
-                        listener['callback'](event)
+        if 'rooms' in response:
```
Comment (anoadragon453) [inline matrix_client/client.py:595]: We can reduce the indentation a bit by making use of Python's `.get()` operator for lists, e.g:

```python
rooms = response.get("rooms", {})
```

which will set `rooms` to an empty dict if not found in `response`.

File: matrix_client/client.py [hunk 2]
```diff
@@ -583,60 +583,68 @@ def _sync(self, timeout_ms=30000):
...
+            if 'invite' in response['rooms']:
```
Comment (anoadragon453) [inline matrix_client/client.py:596]: The same can apply here and below.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
This looks functional, just a couple stylistic notes.

Also note that this library is no longer actively maintained. You may consider switching to another Python library for client actions, such as https://github.com/poljar/matrix-nio.

File: matrix_client/client.py (hunk 1 above)
Comment (surakin) [reply matrix_client/client.py]: sounds good

**Comment by surakin** [comment]
> This looks functional, just a couple stylistic notes.
> 
> Also note that this library is no longer actively maintained. You may consider switching to another Python library for client actions, such as https://github.com/poljar/matrix-nio.

I know :) I only wanted to keep the home assistant matrix integration functional. They should switch to matrix-nio anyway.

**Review by anoadragon453 — APPROVED** [review]
Thanks for making those changes, and yes that's understandable!

**Pull request merged**